Cognoma.py

```diff
@@ -0,0 +1,656 @@
+
+        # Find all TP53 mutated examples and their gene expressions
+        SNPabsent = np.array(rightDomain[0, :])
+        SNPabsentExp = np.array(leftDomain[0, :])
+
+        for x in range(rightDomain.shape[0]):
+            if rightDomain[x, self.mutationID] == 1:
+                SNPabsent = np.vstack((SNPabsent, rightDomain[x, :]))
+                SNPabsentExp = np.vstack((SNPabsentExp, leftDomain[x, :]))
+
+        # Pick a random grouping of these examples
+        randIndexes = np.random.randint(
+            0, min(SNPpresent.shape[0], SNPabsent.shape[0]) - 1,
+            (n,))
+        SNPpresent = SNPpresent[1:]
+        SNPpresentExp = SNPpresentExp[1:]
+        SNPpresent = SNPpresent[randIndexes]
+        SNPpresentExp = SNPpresentExp[randIndexes]
+
+        SNPabsent = SNPabsent[1:]
+        SNPabsentExp = SNPabsentExp[1:]
+        SNPabsent = SNPabsent[randIndexes]
+        SNPabsentExp = SNPabsentExp[randIndexes]
+
+        # Predict based on the mutations
+        (SNPpresentToExp, _) = modelHandle.rightToLeftModel.predict(
+                                                              SNPpresent)
+
+        # Find the Mean Squared Error
+        SNPpresentToExpMSE = np.square(np.subtract(
+            SNPpresentExp, SNPpresentToExp)).mean(axis=0)
+
+        # Create arrays for induced samples
+        InducedMutation = copy.copy(SNPpresent)
+        InducedMutationExp = copy.copy(SNPpresentExp)
+
+        # Induce a TP53 mutation
+        InducedMutation[:, self.mutationID] = 1
+
+        # Predict based on this mutation
+        (InducedMutationToExp, _) = modelHandle.rightToLeftModel.predict(
+                                                             InducedMutation)
+
+        # Find the Mean Squared Error
+        InducedMutationToExpMSE = np.square(np.subtract(
+            InducedMutationExp, InducedMutationToExp)).mean(axis=0)
+
+        # Run a t test on the MSE's
+        ttest_results = scipy.stats.ttest_ind(
+            SNPpresentToExpMSE, InducedMutationToExpMSE)
+        t_stat = ttest_results.statistic
+        p_val = ttest_results.pvalue
+
+        # Print a table with the t test results
+        table_data = dict(values=[[t_stat], [p_val]])
+        table_labels = dict(values=['Stat', 'PVal'])
+        table = [go.Table(cells=table_data, header=table_labels)]
+
+        py.offline.plot(table,
+                        filename=os.path.join('Output',
+                                              params.dataSetInfo.name,
+                                              f't_test_{str(params.outputNum)}'
+                                              ))
+        #######################################################################
+        # For Scatter Plot of the differentially expressed genes between
+        # real TP53 wildtype vs. real TP53 mutated and
+        # real vs. synthetic TP53 data
+
+        # Induce wiltype TP53
+        InducedSNPpresent = copy.copy(SNPabsent)
```

Comment:
SNPabsent should have zeroes already no?

Reply:
I thought that SNPabsent should have 1, and SNPpresent should have 0.